src/ua.c

```diff
@@ -932,6 +932,48 @@ int ua_uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 }
 
 
+static bool uri_host_local(const struct uri *uri)
+{
+
+	const char *hostv[] = {
+		"localhost",
+		"127.0.0.1",
+		"::1"
+	};
+	int afv[2] = {AF_INET, AF_INET6};
+	const struct sa *sal;
+	struct sa sap;
+	size_t i;
+	int err;
+
+	if (!uri)
+		return false;
+
+	for (i=0; i<ARRAY_SIZE(hostv); i++) {
+
+		if (!pl_strcmp(&uri->host, hostv[i]))
+			return true;
+	}
+
+	for (i=0; i<ARRAY_SIZE(afv); i++) {
```

Comment:
at this point, the address family is already known in `uri->af`.

so there should be no need to check both address families, just use uri->af and lookup
the matching address.